lib/traject/psulib_config.rb

```diff
@@ -193,6 +193,27 @@
   accumulator.uniq!
 end
 
+# Genre Fields
+## Main genre
+to_field 'genre_tsim', extract_marc('650|*0|v:655|*0|abcvxyz:655|*7|abcvxyz')
+
+## Genre facet (sidebar)
+to_field 'genre_facet_ssim' do |record, accumulator|
+  genres = process_genre(record, '650|*0|v:655|*0|a:655|*7|a')
+  accumulator.replace(genres)
+  accumulator.uniq!
```

Comment:
I guess a further question might be, should this `uniq` live in the method? Gets called both times following the client code call for it...

Reply:
Ha, technically you could make this one line too...

```
  accumulator.replace(process_genre(record, '655|*0|abcvxyz:655|*7|abcvxyz')).uniq!
```

Probably to compact though?

Reply:
`accumulator.replace(genres).uniq!` I like this one, I basically did not think of doing it in one line. looks way better. I will update 👍

Reply:
in terms of why I used uniq!, since we are including both `fast` and `lcgft` headings for `655`, there are sometimes dupes. eg. https://blackcat01qa.libraries.psu.edu/catalog/22090542 has

```
650    0 a| Orchestral music v| Scores.
650    7 a| Orchestral music x| Scores. 2| fast 0| (OCoLC)fst01047298
655    7 a| Art music. 2| lcgft
655    7 a| Scores. 2| lcgft
655    7 a| Art music. 2| fast 0| (OCoLC)fst01920007
655    7 a| Scores. 2| fast 0| (OCoLC)fst01692898
```

Reply:
Cool thanks.